hist/hist/src/TGraph.cxx

```diff
@@ -824,12 +824,16 @@ void TGraph::Draw(Option_t *option)
    }
 
    // If no option is specified, it is defined as "alp" in case there
-   // no current pad or if the current pad as no axis defined.
-   if (!option || !strlen(option)) {
+   // no current pad or if the current pad as no axis defined and if there is
+   // no default option set using TGraph::SetOption. If fOption is set using
+   // TGraph::SetOption, it is used as default option.
+   if ((!option || !strlen(option))) {
+      Option_t *topt = (fOption.Data() && strlen(fOption.Data())) ? fOption.Data() : "alp";
```

Comment:
You can just use here `fOption.IsNull()` instead of checking the length of the contained const char *  in String